tests/test_modbus_datatype_errors.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+import json
+
+from zabbix_get import zabbix_get
+
+
+class ModbusDatatypeErorrsTestCase(unittest.TestCase):
+
+    host = "172.16.238.2:5020"
+
+    def test_modbus_datatype_no_multiplier_sign(self):
+        formula = "2int16"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: There must be "*" sign after a multiplier and before a type in the datatype expression.')
+
+    def test_modbus_datatype_negative_multiplier(self):
+        formula = "-2*int16"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Multiplier must be positive in the datatype exression.')
+
+    def test_modbus_datatype_unknown(self):
+        formula = "2*bad"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Invalid type in the datatype expression.')
+
+    def test_modbus_datatype_unexpected_end_0(self):
+        formula = "2*int16+"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Unexpected end of the datatype expression.')
+
+    def test_modbus_datatype_skip_0(self):
+        formula = "skip"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Invalid type in the datatype expression.')
+
+    def test_modbus_datatype_skip_1(self):
+        formula = "2*skip"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Invalid type in the datatype expression.')
+
+    def test_modbus_datatype_skip_2(self):
+        formula = "'  skip   '"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Invalid type in the datatype expression.')
+
+    @unittest.skip("This test currently breaks the module")
+    def test_modbus_datatype_whitespace_0(self):
+        formula = "'     '"
+        key = "modbus_read_registers["+self.host+",1,2,3,"+formula+"]"
+        self.assertEqual(zabbix_get(key), 'ZBX_NOTSUPPORTED: Invalid type in the datatype expression.')
+    
+    @unittest.skip("This test currently breaks the module")
+    def test_modbus_no_datatype_for_read_registers(self):
```

Comment:
In case of empty datatype param - defaults should be used:
> otherwise, defaults will be used:
MODBUS_BIT if modbus function 1 or 2.
MODBUS_UINT16 if modbus_function 3 or 4.
But currently providing empty datatype just makes module crash.

Reply:
Should be fixed by 5ff5080899fc9d2b462bdf274c69a37013a449c5, please double check.

Reply:
also these units test for empty datatypes:  `modbus_no_datatype_for_read_coils`, `modbus_no_datatype_for_read_registers`